### PR TITLE
Implement Ability to Extract Q-factors from Eigenmode Simulations

### PR DESCRIPTION
**Pull request by SamarthH**
### Did you add tests to cover your changes (yes/no)?
yes
### Did you update the documentation accordingly (yes/no)?
no
### Did you read the CONTRIBUTING document (yes/no)?
yes
### Summary
This pull request implements the ability to render and use resistive ports in Eigenmode simulations in order to find the Q-factors of resonators and to simulate other dissipative effects.

### Details and comments
- Included a variable in QAnsysRenderer when execute_design is called to store the solution type
- After that, when create_ports is called by hfss_renderer, choose to create RLC ports if the solution type is Eigenmode
- Added tutorial to showcase how-to.

### CODE REVIEW

**Comment by SamarthH** [comment]
For some reason, git is not diffing the files properly, and hence the commit looks much much larger than it is. It is just a few lines of code in execute_design and create_ports.

**Comment by gsteele13** [comment]
i would love to try this out!

**Comment by SamarthH** [comment]
Sorry for the delay. I shall also push the extra meshing part in a moment.

**Comment by SamarthH** [comment]
I think that this is all that I have to implement at the moment. Please review this and let me know of any necessary changes @zlatko-minev @ThomasGM4 . Thanks a lot!

**Comment by ThomasGM4** [comment]
Thanks for working on this. Will look through the code shortly, might take a little bit extra just because of the way the diff is for the entire file instead of the code you changed (I am also not sure why it did it like that).

**Comment by zlatko-minev** [comment]
Ditto @ThomasGM4 , excited to look over, thanks for the contribution work!

File: qiskit_metal/renderers/renderer_ansys/hfss_renderer.py [hunk 1]
```diff
@@ -1,1066 +1,1075 @@
...
+        """
+        ansys_options = dict(transparency=0.0)
+        qcomp = self.design._components[qgeom['component']].name
+        qc_elt = get_clean_name(qgeom['name'])
+        qc_name = 'Lj_' + qcomp
+        inductor_name = f'{qc_name}{QAnsysRenderer.NAME_DELIM}{qc_elt}'
+        # Draw rectangle for inductor.
+        self.logger.debug(f'Drawing a rectangle: {inductor_name}')
+        poly_ansys = self.modeler.draw_rect_corner([xmin, ymin, z], xmax - xmin,
+                                                   ymax - ymin, z,
```
Comment (ThomasGM4) [inline qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:382]: good catch

**Review by ThomasGM4 — CHANGES_REQUESTED** [review]
You will want to clear all the outputs for your demo notebook, all the shapely warnings are still present. (4.18 Analyse a Resonator with Ports.ipynb)

Couple comments in the code itself but it looks good (not sure why it did so much line replacement, used BeyondCompare for sanity check, and most the replaced lines was just due to " -> '

File: qiskit_metal/renderers/renderer_ansys/hfss_renderer.py [hunk 2]
```diff
@@ -1,1066 +1,1075 @@
...
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
```
[the remark this answers is not in the document]
Comment (SamarthH) [reply qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:14]: Sorry, I don't know how that happened. Did not intend to do that. Shall add that back

File: qiskit_metal/renderers/renderer_ansys/hfss_renderer.py [hunk 3]
```diff
@@ -1,1066 +1,1075 @@
...
+                                                       **dict(transparency=0.0))
+            axis = 'x' if abs(x1 - x0) > abs(y1 - y0) else 'y'
+
+            # Add port. Adds RLC port if it is an eigenmode simulation. Otherwise, adds a lumped port.
+            if self.solution_type != 'eigenmode':
+                poly_ansys.make_lumped_port(axis,
+                                            z0=str(impedance) + 'ohm',
+                                            name=f'LumpPort_{qcomp}_{pin}')
+                self.modeler.rename_obj(poly_ansys, port_name)
+            else:
```
[the remark this answers is not in the document]
Comment (SamarthH) [reply qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:234]: There is indeed one possible case, which is to simulate a parallel plate capacitor (like the one used for a lot of lumped element JPA setups).

File: qiskit_metal/renderers/renderer_ansys/hfss_renderer.py (hunk 1 above)
Comment (SamarthH) [reply qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:382]: Thanks

File: qiskit_metal/renderers/renderer_ansys/hfss_renderer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SamarthH) [reply qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:14]: I have re-added the Chalmers comment. Please take a look

[the remark this answers is not in the document]
Comment (ThomasGM4) [reply qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:14]: Sure you can add your name after that line too

[the remark this answers is not in the document]
Comment (ThomasGM4) [reply qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:14]: Looks good to merge after that.

[the remark this answers is not in the document]
Comment (SamarthH) [reply qiskit_metal/renderers/renderer_ansys/hfss_renderer.py:14]: Done!

**Review by ThomasGM4 — APPROVED** [review]

**Comment by SamarthH** [comment]
@ThomasGM4 : Could you please merge the commits into the main branch? Then I would close the pull request.

**Pull request merged**